qiskit_dynamics/pulse_simulator/pulse_simulator.py

```diff
@@ -0,0 +1,526 @@
+        if target is None:
+            target = Target()
+
+        # add default simulator measure instructions
+        instruction_schedule_map = target.instruction_schedule_map()
+        for qubit in self.options.subsystem_labels:
+            if not instruction_schedule_map.has(instruction="measure", qubits=qubit):
+                with pulse.build() as meas_sched:
+                    pulse.acquire(
+                        duration=1, qubit_or_channel=qubit, register=pulse.MemorySlot(qubit)
+                    )
+
+            instruction_schedule_map.add(instruction="measure", qubits=qubit, schedule=meas_sched)
+
+        self._target = target
+
+    def _default_options(self):
+        return Options(
+            shots=1024,
+            solver=None,
+            solver_options={},
+            subsystem_dims=None,
+            subsystem_labels=None,
+            meas_map=None,
+            normalize_states=True,
+            initial_state="ground_state",
+            meas_level=MeasLevel.CLASSIFIED,
+            max_outcome_level=1,
+            memory=True,
+            seed_simulator=None,
+            experiment_result_function=default_experiment_result_function,
+        )
+
+    def set_options(self, **fields):
+        """Set options for PulseSimulator."""
+
+        validate_subsystem_dims = False
+
+        for key, value in fields.items():
+            if not hasattr(self._options, key):
+                raise AttributeError(f"Invalid option {key}")
+
+            if key == "initial_state":
+                if value != "ground_state" and not isinstance(value, (Statevector, DensityMatrix)):
+                    raise QiskitError(
+                        """initial_state must be either "ground_state",
+                        or a Statevector or DensityMatrix instance."""
+                    )
+            elif key == "meas_level" and value != 2:
+                raise QiskitError("Only meas_level == 2 is supported by PulseSimulator.")
+            elif key == "max_outcome_level":
+                if (value is not None) and (not isinstance(value, int) or (value <= 0)):
+                    raise QiskitError("max_outcome_level must be a positive integer or None.")
+            elif key == "experiment_result_function" and not callable(value):
+                raise QiskitError("experiment_result_function must be callable.")
+
+            if key == "solver":
+                self._set_solver(value)
+                validate_subsystem_dims = True
+            else:
+                if key == "subsystem_dims":
+                    validate_subsystem_dims = True
+                self._options.update_options(**{key: value})
+
+        # perform additional validation if certain options were modified
+        if (
+            validate_subsystem_dims
+            and np.prod(self._options.subsystem_dims) != self._options.solver.model.dim
+        ):
+            raise QiskitError(
```

Comment:
Might be good to later add our own error class subclass for using in the backend.